python/src/aiconfig/eval/common.py

```diff
@@ -0,0 +1,61 @@
+
+from typing import Any, Generic
+from abc import abstractmethod
+from typing import Any, Generic, Protocol, TypeVar
+
+import lastmile_utils.lib.core.api as cu
+from pydantic import root_validator
+
+
+
+T_InputDatum = TypeVar("T_InputDatum", contravariant=True)
+T_OutputDatum = TypeVar("T_OutputDatum", contravariant=True)
```

Comment:
I'm confused, why are we declaring `T_InputDatum` to be bound to itself? https://stackoverflow.com/a/61737894

In general what does `A = TypeVar("A")` do